_posts/2016-12-06-the-state-of-babel.md

```diff
@@ -0,0 +1,460 @@
+
+[motiz88/babel-smoke-tests](https://github.com/motiz88/babel-smoke-tests) is a good start! It would also be great if there's already something like this out there!
+
+### Program Analysis
+
+- [Program Slicing](https://en.wikipedia.org/wiki/Program_slicing) with [slice-js](http://slides.com/kentcdodds/faster-javascript#/)
+
+There was a great talk by [Alan Shreve](https://twitter.com/inconshreveable) called ["Idealized Commit Logs: Code Simplification via Program Slicing"](https://www.youtube.com/watch?v=dSqLt8BgbRQ) which inspired @kentcdodds to try it out in JavaScript.
+
+The general idea is that we have a lot of tools to help us write code but not a lot for helping us understand/read code. You can think of code-slicing as a form of targeted dead-code elimination.
+
+![slice-js](https://s3.amazonaws.com/media-p.slid.es/uploads/55780/images/3189688/Screen_Shot_2016-11-02_at_11.46.25_PM.png)
+
+A program slice basically cuts away from the source code the code that isn't used for a test case that you run. If there are lots of if statements and loops that aren't run during your usecase then it won't show up in the program slice.
+
+- Semantic (AST aware) Grepping tool
+
+Similar to [graspjs](http://www.graspjs.com/), I think it would be interesting to be able to do a find-replace with the input being an AST. It would allow us to create other analysis tools: the ability to find all IIFE's in our code, the number of times a method is called, or even how many Classes we have in our codebase.
+
+### `babel --settings`
+
+This command would print out all info (also when erroring). It would also include performance metrics on how long each plugin takes.
+
+- [`babel --settings`](https://github.com/babel/babel/issues/2960)
+
+### Parser Unity
+
+There have also been some discussions around parser/AST unity, in [TheLarkInn/js-parser-discussions](https://github.com/TheLarkInn/js-parser-discussions) and previously with [ESTree](https://github.com/estree/estree).
+
+Unfortunately with Babel 6, we have "forked" and have a few differences in [our AST](https://github.com/babel/babylon#output) than ESTree. Babel inherently will want to support stage x features while other parsers may only want to support stage 4 features. We all might prioritize certain things more than others regarding spec compliancy, performance, stage x features, error messages, extensibility, releases, etc. However it's important for us to be open to breaking changes that may lead to better interop and a better community.
+
+### Sweet.js Interop
+
+Previous [issue](https://github.com/babel/babel/issues/568#issuecomment-71716260). Maybe we can just figure out how to have better interop instead?
+
+### Others?
+
+Anything not already brought up here? Please send us a tweet [@babeljs](https://twitter.com/babeljs), a message on our slack (join at [http://slack.babeljs.io/](slack.babeljs.io), comment on this post, or create an discussion issue in our repo!
```

Comment:
I’d love to see the issue of supporting old Node.js versions addressed in this post (even if it’s just a link to the discussion at https://github.com/babel/babel/issues/4315).